dotnet/src/Connectors/Connectors.AI.OpenAI/ImageGeneration/AzureOpenAIImageGeneration.cs

```diff
@@ -0,0 +1,153 @@
+    /// <summary>
+    /// Constants of success status
+    /// </summary>
+    private const string SucceededStatus = "SUCCEEDED";
+
+    /// <summary>
+    /// Maximum retry times for the Azure Dall-E API
+    /// </summary>
+    private readonly int _maxRetryCount;
+
+    /// <summary>
+    /// Default retry wait time in seconds
+    /// </summary>
+    private readonly int _defaultRetryAfter;
+
+    /// <summary>
+    /// Azure OpenAI REST API endpoint
+    /// </summary>
+    private readonly string _endpoint;
+
+    /// <summary>
+    /// Azure OpenAI API key
+    /// </summary>
+    private readonly string _apiKey;
+
+    /// <summary>
+    /// Create a new instance of Azure OpenAI image generation service
+    /// </summary>
+    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
+    /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
+    /// <param name="logger">Application logger</param>
+    /// <param name="maxRetryCount">Maximum number of retries for HTTP requests.</param>
+    /// <param name="defaultRetryAfter">Default retry wait time in seconds for HTTP requests.</param>
+    public AzureOpenAIImageGeneration(string endpoint, string apiKey, HttpClient? httpClient = null, ILogger? logger = null, int maxRetryCount = 5, int defaultRetryAfter = 6) : base(httpClient, logger)
+    {
+        this._endpoint = $"{endpoint}openai/images/generations:submit?api-version=2023-06-01-preview";
+        this._apiKey = apiKey;
+        this._maxRetryCount = maxRetryCount;
+        this._defaultRetryAfter = defaultRetryAfter;
+    }
+
+    /// <inheritdoc/>
+    public async Task<string> GenerateImageAsync(string description, int width, int height, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(description);
+        if (width != height || (width != 256 && width != 512 && width != 1024))
+        {
+            throw new ArgumentOutOfRangeException(nameof(width), width, "OpenAI can generate only square images of size 256x256, 512x512, or 1024x1024.");
+        }
+
+        var requestBody = Json.Serialize(new ImageGenerationRequest
+        {
+            Prompt = description,
+            Size = $"{width}x{height}",
+            Count = 1
+        });
+
+        var result = await this.GetImageGenerationAsync(this._endpoint, requestBody, cancellationToken).ConfigureAwait(false);
+        if (result.Result == null)
+        {
+            throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "Response not contains result");
+        }
+
+        return result.Result.Images[0].Url;
+    }
+
+    private async Task<AzureImageGenerationResponse> GetImageGenerationAsync(string url, string? requestBody, CancellationToken cancellationToken = default)
```

Comment:
This method has multiple responsibilities. It would be good to extract those responsibilities into separate methods with clear names, distinguishing the POST and GET requests.